psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1264,241 +1264,6 @@ std::shared_ptr<std::vector<double>> ESPPropCalc::compute_esp_at_nuclei(bool pri
-        if (same_dens_) {
-            Da = Da_so_;
-            Db = Da;
-        } else {
-            Da = Da_so_;
-            Db = Db_so_;
-        }
-    } else {
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole XX", basisset_->nbf(), basisset_->nbf()));
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole XY", basisset_->nbf(), basisset_->nbf()));
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole XZ", basisset_->nbf(), basisset_->nbf()));
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole YY", basisset_->nbf(), basisset_->nbf()));
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole YZ", basisset_->nbf(), basisset_->nbf()));
-        qpole_ints.push_back(std::make_shared<Matrix>("AO Quadrupole ZZ", basisset_->nbf(), basisset_->nbf()));
-        std::shared_ptr<OneBodyAOInt> aoqOBI(integral_->ao_quadrupole());
-        aoqOBI->set_origin(origin_);
-        aoqOBI->compute(qpole_ints);
-        if (same_dens_) {
-            Da = wfn_->matrix_subset_helper(Da_so_, Ca_so_, "AO", "D");
-            Db = Da;
-        } else {
-            Da = wfn_->matrix_subset_helper(Da_so_, Ca_so_, "AO", "D alpha");
-            Db = wfn_->matrix_subset_helper(Db_so_, Cb_so_, "AO", "D beta");
-        }
-    }
-
-    if (verbose)
-        for (int n = 0; n < 6; ++n) qpole_ints[n]->print();
-
-    // Each multipole integral needs to be dotted with the SO Density
-    Vector qe(6);
-    qe[0] = Da->vector_dot(qpole_ints[0]) + Db->vector_dot(qpole_ints[0]);
-    qe[1] = Da->vector_dot(qpole_ints[1]) + Db->vector_dot(qpole_ints[1]);
-    qe[2] = Da->vector_dot(qpole_ints[2]) + Db->vector_dot(qpole_ints[2]);
-    qe[3] = Da->vector_dot(qpole_ints[3]) + Db->vector_dot(qpole_ints[3]);
-    qe[4] = Da->vector_dot(qpole_ints[4]) + Db->vector_dot(qpole_ints[4]);
-    qe[5] = Da->vector_dot(qpole_ints[5]) + Db->vector_dot(qpole_ints[5]);
-
-    // Add in nuclear contribution
-    if (!transition) {
-        SharedVector nquad = QuadrupoleInt::nuclear_contribution(mol, origin_);
-        qe[0] += nquad->get(0, 0);
-        qe[1] += nquad->get(0, 1);
-        qe[2] += nquad->get(0, 2);
-        qe[3] += nquad->get(0, 3);
-        qe[4] += nquad->get(0, 4);
-        qe[5] += nquad->get(0, 5);
-    }
-
-    // Print multipole components
-    double dfac = pc_dipmom_au2debye * pc_bohr2angstroms;
-    if (print_output) {
-        outfile->Printf("  %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
-        outfile->Printf("    XX: %10.4lf     YY: %10.4lf     ZZ: %10.4lf\n", qe[0] * dfac, qe[3] * dfac, qe[5] * dfac);
-        outfile->Printf("    XY: %10.4lf     XZ: %10.4lf     YZ: %10.4lf\n", qe[1] * dfac, qe[2] * dfac, qe[4] * dfac);
-        outfile->Printf("\n");
-    }
-
-    double dtrace = (1.0 / 3.0) * (qe[0] + qe[3] + qe[5]);
-    if (print_output) {
-        outfile->Printf("  Traceless %sQuadrupole Moment: [D A]\n", (transition ? "Transition " : ""));
```

Comment:
is traceless quadrupole handled by the general multipole code even though non-systematic?

Reply:
Ugh, traceless quadrupole is _not_ handled by the general code. Although I'm not familiar with multipole theory, it looks like traceless multipoles are defined for all orders beyond dipole... I am seriously considering spending the weekend reading up so I have an idea of how to implement this in general.